### DISCUSSION TITLE
#4 📑 Weekly Report 4

### DISCUSSION THREAD

**Discussion by fehlend0** [SE-TINF22B6/Statify#41] [Blog]
Hi there 👋

Here's what we've accomplished for the project this week.

## Landing Page Development
We've been busy this week bringing our Figma mockup to life using React.js. This means that we're turning that fancy design you saw into an actual web page and this is how it looks like now:
 
![photo_2023-11-08_17-15-29](https://github.com/SE-TINF22B6/Statify/assets/106541016/06eaffc9-0e3c-4db0-a768-13f150dfac38)

## Unit Testing Frameworks
We've also been exploring the frameworks for unit testing. We chose two frameworks - jest and junit. To begin with, we created a special test for the button-component on our web page to make sure that when users click buttons, everything behaves as it should. Further unit tests are on their way!

## CI Workflows Setup
We've created a [workflow](https://github.com/SE-TINF22B6/Statify/blob/56d3aa229d150cc2ef4f0fd5d8d89ac597ab35ca/.github/workflows/node.js.yml) that does all the heavy lifting - installing dependencies, saving them for future use, building our code, and running tests on different versions of Node. As for [our other workflow](https://github.com/SE-TINF22B6/Statify/blob/56d3aa229d150cc2ef4f0fd5d8d89ac597ab35ca/.github/workflows/maven.yml), it is designed to automate the build and testing process in the backend. It ensures that the project is built using JDK 21, and it also optionally updates the project's dependency graph for quality monitoring purposes. The workflow is triggered when code is pushed to the "main" or "develop" branches or when pull requests are made against the "main" branch. 

## Activity Diagrams Update
We're keeping [our project documentation](https://github.com/SE-TINF22B6/Statify/blob/56d3aa229d150cc2ef4f0fd5d8d89ac597ab35ca/Documentation/srs.md) up to date, including [the activity diagrams](https://github.com/SE-TINF22B6/Statify/tree/56d3aa229d150cc2ef4f0fd5d8d89ac597ab35ca/Documentation/diagrams), which were updated after the last week feedback.

That's it for this week. If you have any questions or need more details, feel free to ask!

Your Statify Team

**Comment by LenVej** [comment]
Dear Statify Team,

It's clear that you're making significant progress on your project and I really like the clean design of your landing page and the used colors. 
Your CI workflows are well-structured and seem to cover key aspects of your project. The Node.js CI workflow appears to efficiently handle Node.js dependencies, building, and testing. The Java CI with Maven workflow similarly covers essential steps for building Java projects. The use of GitHub Actions is a good choice as it's a widely used and well-integrated CI/CD tool.
You could consider adding explanatory notes to your workflow configuration files. While the workflows are clear for someone familiar with GitHub Actions, adding comments or explanations for each step can make it more accessible to newcomers

Overall, you seem to be on the right track with your CI setup and unit testing framework choices and I'm excited to see your project grow.

Best Regards,
Your CookHub Team

**Comment by AndresMasis** [comment]
Hi.
First of all I want to congratulate you for your front-end, which is very appealing visually.
Then, I find interesting your decision of using 2 testing tools. It would be nice if you can explain why did you take that decision.
I also congratulate you for your CI workflow. Indeed is very robust and it does meet the expectations of automating several processes to ensure the continous availability of the working environment.

Have a nice week,
CookPit Team
